preview-maker.js

```diff
@@ -79,33 +101,127 @@ const makePrev = async (themeBuffer, themeName, themeAuthor, template) => {
+    }
+    if (colors.length != 0) {
+        var colorsquantity = {}
+        var max = 0
+        for (const colornow of colors) {
+            if (colornow != 0) {
+                if (colorsquantity[colornow] == null) {colorsquantity[colornow] = 0}
+                    colorsquantity[colornow] += 1
+                if (colorsquantity[colornow] > max) {
+                    accentHue = colornow
+                    max = colorsquantity[colornow]
+                }
+            }
+        }
+        var colorsHue = []
+        for (colornow in colorsquantity) {
+            if (colorsquantity[colornow] == max) {
+                colorsHue.push(colornow);
+            }
+        }
+        if (colorsHue.length > 1) {
+            var minDifference = 360
+            for (const colorHue of colorsHue) {
+                for (const hue of colorsHue) {
+                    if (hue != colorHue) {
+                        if (Math.abs(hue - colorHue) < minDifference) {
+                            minDifference = Math.abs(hue - colorHue)
+                            accentHue = (hue + colorHue) / 2
+                        }
+                    }
+                }
+            }
+        }
+    } else {
+        accentHue = rgbToHsl(theme['chats_actionBackground'] || defaultVariablesValues['chats_actionBackground']).hue
+    }
+    for (const outBubbleGradientelement of getElementsByClassName(preview, "chat_outBubbleGradient")){
+        const chat_outBubble = theme[`chat_outBubble`] || defaultVariablesValues["chat_outBubble"];
+        fill(outBubbleGradientelement, theme[`chat_outBubbleGradient`] || chat_outBubble);
+    }
+    for (const PreviewBackLinear of getElementsByClassName(preview, "PreviewBackLinear")) {
+        fillHSL(PreviewBackLinear, `hsl(${rgbToHsl(theme[`chat_outBubbleGradient`]).hue}, 100%, 90%)`);
+    }
+    for (const PreviewBackLinearShadow of getElementsByClassName(preview, "PreviewBackLinearShadow")) {
+        fillHSL(PreviewBackLinearShadow, `hsl(${rgbToHsl(theme[`chat_outBubble`] || defaultVariablesValues["chat_outBubble"]).hue}, 100%, 90%)`);
+    }
+    for (const PreviewBack of getElementsByClassName(preview, "PreviewBack")) {
+        fillHSL(PreviewBack, `hsl(${accentHue}, 100%, 90%)`);
+    }
+    for (const ChatShadow of getElementsByClassName(preview, "ChatShadow")) {
+        fillHSL(ChatShadow, `hsl(${accentHue}, 100%, 2%)`);
+    }
+    if (theme[`chat_outBubbleGradient`] != undefined) {
+        for (const PreviewBack of getElementsByClassName(preview, "PreviewBack")){
+            fill(PreviewBack, {red:0,green:0,blue:0,alpha:0});
+        }
+    }
+    for (const avatar of avatars){
+        const windowBackgroundWhite = theme['windowBackgroundWhite'] || defaultVariablesValues['windowBackgroundWhite']
+        if (Math.abs(theme[avatar].red - windowBackgroundWhite.red) + Math.abs(theme[avatar].green - windowBackgroundWhite.green) + Math.abs(theme[avatar].blue - windowBackgroundWhite.blue) < 15){
```

Comment:
This code seems to calculate the distance between two colors, but this is not the correct way to do it. Visualizing, you want to check that one color is inside a sphere with the other color at its center:

![image](https://user-images.githubusercontent.com/10610844/100545575-0cb78d00-328f-11eb-9468-63af1db0e2bc.png)

but you actually check that one color is inside an _octahedron_ with the other color at its center:

![image](https://user-images.githubusercontent.com/10610844/100545777-20afbe80-3290-11eb-9957-4a107169754d.png)

The formula for calculating the distance between two points is √((x₁ - x₂)² + (y₁ - y₂)² + (z₁ - z₂)²), or, applied to this situation,

```js
Math.hypot(theme[avatar].red - windowBackgroundWhite.red, ..)
```

Given that you calculate it several times, this better be in its own function.